server/src/services/semanticHighlight/highlighters/ContractDefinitionHighlighter.ts

```diff
@@ -9,7 +9,7 @@ export class ContractDefinitionHighlighter extends HighlightVisitor {
   public tokenKinds = new Set([TokenKind.Identifier]);
 
   public enter(nodeWrapper: SlangNodeWrapper): void {
-    const ancestors = nodeWrapper.pathRuleNodes;
+    const ancestors = nodeWrapper.ancestors();
     if (
       nodeWrapper.type === NodeType.Token &&
```

Comment:
I wonder if it is worth it to also start checking for `FieldName` in `HighlightVisitor` implementations. Thoughts?

Reply:
Done in https://github.com/NomicFoundation/hardhat-vscode/pull/545/commits/5be2bb25bb0dd48421780f415a23434eec4b2b92. Not that helpful as it turns out but at least it showed which cases we're missing! I'm not trying to improve their coverage here, we can do that in a follow-up.